library/L1_Peripheral/msp432p401r/system_controller.hpp

```diff
@@ -128,14 +191,91 @@ class SystemController final : public sjsu::SystemController
-    uint8_t frequency_select;
-    /// The 10-bit signed tuning value used to obtain the target DCO frequency.
-    int16_t tuning_value;
+    /// Configurations for the auxiliary clock module.
+    struct AuxiliaryClock_t
+    {
+      /// @note Only the reference clock, very low frequency clock, or the low
+      ///       frequency clock can be used as the clock source.
+      Oscillator clock_source = Oscillator::kReference;
+      /// Clock divider for the auxiliary clock.
+      ClockDivider divider = ClockDivider::kDivideBy1;
+    };
+
+    /// Configurations for the master clock module. This clock module is also
+    /// used to drive the CPU clock.
+    struct MasterClock_t
+    {
+      /// On reset, the master clock is driven by the digitally controlled
+      /// clock.
+      Oscillator clock_source = Oscillator::kDigitallyControlled;
+      /// Clock divider for the master clock.
+      ClockDivider divider = ClockDivider::kDivideBy1;
+    };
+
+    /// Configurations for the subsystem master clock module.
+    struct SubsystemMasterClock_t
+    {
+      /// On reset, the subsystem master clocks are driven by the DCO clock.
+      ///
+      /// @note The specified clock source is used to drive both the subsystem
+      ///       master clock and the low speed subsystem master clock.
+      Oscillator clock_source = Oscillator::kDigitallyControlled;
+      /// Clock divider for the subsystem master clock.
+      ClockDivider divider = ClockDivider::kDivideBy1;
+      /// Clock divider for the low speed subsystem master clock.
+      ClockDivider low_speed_divider = ClockDivider::kDivideBy1;
+    };
+
+    /// Configurations for the backup clock module.
+    struct BackupClock_t
+    {
+      /// Clock source for the backup clock.
+      ///
+      /// @note Only the reference clock or the low frequency clock can be used
+      ///       as the clock source.
+      Oscillator clock_source = Oscillator::kReference;
+    };
+
+    /// Configurations for the reference clock. The reference clock is
+    /// configurable to output 32.768 kHz or 128 kHz.
+    struct ReferenceClock_t
+    {
+      /// When frequency_select = 0b0, the reference clock outputs 32.768 kHz.
+      /// When frequency_select = 0b1, the reference clock outputs 128 kHz.
+      uint8_t frequency_select = 0b0;
+    };
+
+    /// Configurations for the digitally controlled (DCO) clock module.
+    struct DigitallyControlledClock_t
+    {
+      /// On reset, the digitally controlled clock is used to drive the master
+      /// clock and the subsystem master clocks. This value should be set to
+      /// false if an alternate clock source is used to drive those clocks.
+      bool enabled = true;
+      /// The target DCO output frequency. On reset, DCO outputs 3 MHz.
+      units::frequency::hertz_t frequency = 3_MHz;
+    };
+
+    //! @cond Doxygen_Suppress
+    AuxiliaryClock_t auxiliary              = {};
```

Comment:
```suggestion
    struct {
      /// @note Only the reference clock, very low frequency clock, or the low
      ///       frequency clock can be used as the clock source.
      Oscillator clock_source = Oscillator::kReference;
      /// Clock divider for the auxiliary clock.
      ClockDivider divider = ClockDivider::kDivideBy1;
    } auxiliary              = {};
```

This is something I did for lpc40xx. What do you think? It reduced the amount of code quite a bit. I feel like creating these structures, since they have absolutely no use outside of this class is not necessary. Thus we can create an anonymous structure inline here.